src/roster/views.py

```diff
@@ -1468,7 +1470,93 @@ def remove_social_account(request, *args, **kwargs):
 
     return HttpResponseRedirect(reverse('my_account'))
 
-class StudentDetailsView(LoginRequiredMixin, ThemedPageMixin, SettingsPageMixin, TemplateView):
+def get_book_details(books, period, clusiveStudent, clusiveUser):
+    book_details = []
+    if books:
+        for one_book in books:
+            book = Book.objects.get(pk=one_book['book_id'])
+            customizations = Customization.objects.filter(book=book, periods=period)
+            comp_checks = ComprehensionCheckResponse.objects.filter(user=clusiveStudent, book=book)
+            custom_responses = ComprehensionCheckResponse.get_class_details_custom_responses(book=book, period=period)
+            custom_response = custom_responses.filter(user=clusiveStudent)
+            category_names = []
+            for category in book.reading_level_categories:
+                category_names.append(category.tag_name)
+
+            book_details.append({
+                'book_id': one_book['book_id'],
+                'title': one_book['title'],
+                'author': book.author,
+                'hours': round(one_book['hours'], 1),
+                'last_view': one_book['last_view'],
+                'view_count': one_book['view_count'],
+                'words_looked_up': ', '.join(json.loads(one_book['words_looked_up'])) if one_book['words_looked_up'] else None,
+                'first_version': one_book['first_version'],
+                'last_version': one_book['last_version'],
+                'custom_question': '(' + customizations[0].question + ')' if customizations else None,
+                'custom_response': custom_response[0].custom_response if custom_response else None,
+                'learning': comp_checks[0].get_answer if comp_checks else None,
+                'reading_level': ', '.join(category_names),
+                'is_assigned': one_book['is_assigned'],
+                'version_switched': True if one_book['first_version'] and one_book['first_version'] != one_book['last_version'] else False,
+                'unauthorized': not book.is_visible_to(clusiveUser)
+            })
+    print('=== book_details: ', book_details)
```

Comment:
Was this `print()` for debugging?  If so, remove it.  If not, better to user the `logger` here.